### ISSUE TITLE
Battle Net

### ISSUE THREAD

**Issue by Alexander-Carpenter** [ValveSoftware/Proton#7302]
# Compatibility Report
- Name of the game with compatibility issues: Battle Net
- Steam AppID of the game:

## System Information
- GPU: GeForce GTX 1660 Ti
- Video driver version: nvidia 525.147.05
- Kernel version: 5.15.0-89-generic
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: experimental-8.0-20231129

## I confirm:
- [ X ] that I haven't found an existing compatibility report for this game.
- [ X ] that I have checked whether there are updates for my system available.

## Symptoms 
When attempting to launch Battle Net, it never loads but does not crash. You can skip the login process but you get a blank Battle Net screen. At the log in screen, the symbol continuously is circling.

## Reproduction
Attempt to open Battle Net that was installed.

[steam-17019436345753337856.log](https://github.com/ValveSoftware/Proton/files/13518339/steam-17019436345753337856.log)

**Comment by LittleBigBug** [comment]
Having the same issues on my deck. My dad had done the same on his deck about a week ago, and everything works for him.

My launcher version: 2.27.0.14542 (broken)
His launcher version: 2.26.1.14494 (working)

We are both using Proton 8/Experimental. I tried using Proton GE as well to no avail

dang you blizzard!!

**Comment by lekzz** [comment]
"Adding WINE_SIMULATE_WRITECOPY=1 as an environment variable fixes 14542."

"For anyone using Lutris, you just need to add the flag on Configure > System Options > [Scrolldown to “Game Execution”] > Environment Variables."

"If you're launching with steam, change launch options to WINE_SIMULATE_WRITECOPY=1 %command%"

**Comment by Mr-nUUb** [comment]
I googled for nearly an hour, but cannot find any information on this env variable.
So what does `WINE_SIMULATE_WRITECOPY` do?

EDIT: I would't label this issue "unofficial". Isn't Diablo IV (app ID 2344520) launched through Battle.net?

**Comment by andrewschott** [comment]
> EDIT: I would't label this issue "unofficial". Isn't Diablo IV (app ID 2344520) launched through Battle.net?

Nope, Bnet is not in the foodchain on Steam D4.  Which is why I bought it there.

**Comment by danielw1990** [comment]
Is there a way to re-download the working version?
I forgot to make a backup and bnet deleted t on a recent launch, leaving me without a working version.

**Comment by andrewschott** [comment]
> Is there a way to re-download the working version? I forgot to make a backup and bnet deleted t on a recent launch, leaving me without a working version.

How are you launching BNet?  I am using proton experimental via steam and can play d4 and d2 with zero issue.  Bnet version appears to be the latest (no updates pending) -- 2.28.0.14589

**Comment by ruinedraziel** [comment]
Battle.Net Agent start failing for me a few days ago.

So I found a solution that worked, usingo proton 10.0 beta, and it does work, but Steam MUST be open otherwise I got this message.

![Image](https://github.com/user-attachments/assets/46ed8208-d893-48c9-9e56-08e31e1bc7dd)

Will it be necessary to have steam open on the future?

**Comment by kisak-valve** [comment]
Hello @ruinedraziel, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

**Comment by alasky17** [comment]
@ruinedraziel Also - how are you installing/launching battle.net? Could you give some more detailed steps on the failing setup with steam not running and how to recreate it?

**Comment by ruinedraziel** [comment]
Sure thing, I'm at work, will compile the info when im home. It started when the agent would not work anymore on ge-proton 9.27, tried to reinstall it wont finishe stuck at 45%, using lutris installation. Has already spawn reports in other places (https://www.reddit.com/r/linux_gaming/comments/1kbu0ia/comment/mpxhs0p/).

And if you have installed already the agent wont start unless you are on proton 10, and if you are on proton 10 I get this message if steam is not opened, this is reproducible in my two main systems. both similar configs (desktop and laptop).

I'll also document a crude vm step to step to see if its reproducible that way also. @alasky17 

edit: Right now, if I revert back to proton 9 battle net agent will not work. So im assuming some fix from the 10 beta will be necessary anyway.

**Comment by ruinedraziel** [comment]
@alasky17 Srry for the delay, I removed the battle.net app and tried to reinstall it, Im using Lutris.

When trying to install while using Proton 10.0 Beta I got this error without even seeing the installer:

![Image](https://github.com/user-attachments/assets/dbb4f606-1b9d-4f21-b694-676db729d2a1)
After canceling this message a few times installation proceeds.

After installing, every launch of battle.net give me this message: 

![Image](https://github.com/user-attachments/assets/61675311-a2c4-4c79-8b22-97f5c029d128)

No matter the answer, ok or cancel, battle.net fails to start.

So I tried clean, new VM, Arch linux > steam > installed proton 10.0 > closed steam > installed lutris > installed wine > Picked proton 10 > installed battle.net
same error

![Image](https://github.com/user-attachments/assets/1a41b877-c311-4492-9d32-99a52f4c1198)

but this time installation wont progress any further (probably VM related)
 
I can do this on a bare metal on the weekend and report back.

Here is system information, and lutris logs.

[systemspec.txt](https://github.com/user-attachments/files/20095427/systemspec.txt)
[systeminfo.txt](https://github.com/user-attachments/files/20095428/systeminfo.txt)
[lutrislog.txt](https://github.com/user-attachments/files/20095431/lutrislog.txt)

edit: fixed some typos

**Comment by ruinedraziel** [comment]
Srry, adding up PROTON_LOG=1 with both situations steam closed, and open, cant see a diference on them except for the number of lines.

[steam-open.log](https://github.com/user-attachments/files/20096706/steam-open.log)
[steam-closed.log](https://github.com/user-attachments/files/20096705/steam-closed.log)

**Comment by kisak-valve** [comment]
Hello @ruinedraziel, `System WINEDEBUG: -all` tells us something has set the WINEDEBUG environment variable to make the Proton log unusable for troubleshooting.

**Comment by ruinedraziel** [comment]
@kisak-valve Thx for the feedback, I will look where it's coming from and remove it, probably something on lutris, and redo the logs. Update when I'm home.

Is this non reproducible for you guys? I might have to make a bare bone clean install for testing then.

edit: fixed some typos...

**Comment by ruinedraziel** [comment]
So, my laptop repeat the exact same behaviour, and I found where the `winedebug -all` come from. So here is a actual log from a second pc with the exact same behaviour, so here is the logs. Just clearing up, battle.net work as intended if steam is open.

[proton-steamclosed.log](https://github.com/user-attachments/files/20103604/proton-steamclosed.log)
[proton-steamopen.log](https://github.com/user-attachments/files/20103602/proton-steamopen.log)
[systemreport.txt](https://github.com/user-attachments/files/20103601/systemreport.txt)
[systemspec.txt](https://github.com/user-attachments/files/20103603/systemspec.txt)

**Comment by alasky17** [comment]
@ruinedraziel It would be great if you could try an install without Lutris to remove that variable.  It is possible to manually install battle.net as a non-steam game (I know because I've done it :).  Also yes - VMs can create all sorts of chaos 😅 

Edit: Also - when you say it works great with Steam open -- do you mean if Steam is running in the background but the steam window is closed, it doesn't work?  Or do you mean it only fails if Steam processes are completely shut down (in which case, how do you launch battle.net?)?

**Comment by ruinedraziel** [comment]
> [@ruinedraziel](https://github.com/ruinedraziel) It would be great if you could try an install without Lutris to remove that variable. It is possible to manually install battle.net as a non-steam game (I know because I've done it :). Also yes - VMs can create all sorts of chaos 😅
> 
> Edit: Also - when you say it works great with Steam open -- do you mean if Steam is running in the background but the steam window is closed, it doesn't work? Or do you mean it only fails if Steam processes are completely shut down (in which case, how do you launch battle.net?)?

Yes, steam must be running, it can be minized to tray, don't need to be visible on screen.

Im opening from lutris, I've always used it this way,

I agree that would probably be better off runing b.net from steam already, but if I run it as a non-steam game trough steam the problem probably will disapear, as steam will be running.

And I'm able to use it and play without issues as long as steam is running without changing how I currently use. :)

**Comment by ivyl** [comment]
In Proton 10 we have reworked how our `steam.exe` shim works. This is not the full blown Steam having issues here but just our small helper source of which you can find in the Proton tree. It was rewritten to use the new PE split instead of the old `.exe.so` mechanism and apparently it's fatal for it when it's not run in the Steam environment. **We'll look into that and try to fix it.**

It does query some things via steamapi as well as populated Steam-related registry entries and creates .vdf files for the benefit of mod launchers and similar. Also some Steam games expect to be children of the `steam.exe` process.

**In the meantime** you can just make sure the build you are using outside of Steam doesn't use steam.exe:

```diff
diff --git a/proton b/proton
index 0cd637496bd1..509a36184a6f 100755
--- a/proton
+++ b/proton
@@ -1831,9 +1831,9 @@ class Session:
                     "311210",   # CoD: Black Ops 3
                     "1549250",  # Undecember
                 ]:
-            argv = [g_proton.wine_bin, "c:\\Program Files (x86)\\Steam\\steam.exe"]
+            argv = [g_proton.wine_bin]
         else:
-            argv = [g_proton.wine64_bin, "c:\\windows\\system32\\steam.exe"]
+            argv = [g_proton.wine64_bin]
 
         rc = self.run_proc(adverb + argv + sys.argv[2:] + self.cmdlineappend)
 
```

**Comment by kisak-valve** [comment]
Battle.net not launching

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9753.
@Az0na posted on 2026-05-06T20:05:45:

I got issue today launching battle.net, few days ago it startup fine with Proton Experimental but today it refuses, progress starts I see it flikker in tray/taskbar also see progress start in Misson Center, start 4 process then they disappear again. No errors or anything.

I tried different version of Proton all with same effect, also reinstalled battle.net trough steam as I did before but same result. So I think battle.net app got update that broke it.

My hardware if it matters.
GPU: RTX 2070 SUPER
CPU: Ryzen 9 3900X
Memory: 32 GB
OS: Zorin OS 18.1 Core

**Comment by kisak-valve** [comment]
Hello @Az0na, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by Az0na** [comment]
[steam-17945278634453368832.log](https://github.com/user-attachments/files/27493124/steam-17945278634453368832.log)

https://gist.github.com/Az0na/29bfe7cf2e65f5069cfe4cac2ee61b3e

This enough?

**Comment by kisak-valve** [comment]
https://gist.github.com/Az0na/29bfe7cf2e65f5069cfe4cac2ee61b3e#file-gistfile1-txt-L1133-L1146 and `err:   DxvkInstance::createInstance: Failed to create Vulkan instance` in your Proton log are the sections to focus on. The NVIDIA proprietary driver needs for the driver's running kernel module, 32 bit userspace libraries, and 64 bit userspace libraries are all installed and exactly the same version to work properly. You appear to be lacking the 32 bit userspace libraries.

**Comment by Az0na** [comment]
seems nvidia drivers 595 does not have 32 libraries, I downgraded to 580 and working fine now. used 590 before that worked to so not sure why 595 does not has them. seems be ubuntu issue as I run Zorin OS 18.1 based on ubuntu and I read somewhere cashyos does not have this problem with the drivers. oh well. thanks for the assistance.

**Comment by kisak-valve** [comment]
[Ubuntu 26.04] Battle.net Launcher / Blizzard Agent infinite NtGetContextThread loop and gameoverlayui IPC crash

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9813.
@andrewsuperlegit posted on 2026-05-24T05:01:24:

# Compatibility Report
- Name of the game with compatibility issues:
- Battle.net launcher 


### System Information
* **OS:** Ubuntu 26.04 LTS (x86_64)
* **GPU:** NVIDIA GeForce RTX™ 5070 Laptop GPU and AMD Radeon™ 890M Graphics
* **Driver Version:** NVIDIA-SMI 595.71.05 
* **Kernel Version:** 7.0.0-15-generic
* **Proton Version:**  Tested on Experimental, GE-Proton10-34, and Proton 8.0 and Proton 9.0

### Symptoms
Since upgrading to Ubuntu 26.04, launching the Blizzard Battle.net client as a non-Steam application results in a hang/deadlock before/if the UI initializes. It worked fine on 25.10.

### Diagnostics & Log Analysis
Running with `PROTON_LOG=1` isolates two distinct blocking behaviors:
1. **Steam IPC / Overlay Failure:** `gameoverlayui` crashes completely during initialization due to host-side `glibc` differences on Ubuntu 26.04. This breaks the overlay pipe, drops minidumps, and leaves zombie Wine processes in memory.
2. **Blizzard Agent Race Condition:** Immediately following the `fixme:thread:NtSetInformationThread ThreadPowerThrottling stub!` warning, the Blizzard Agent binary enters an unhandled loop throwing thousands of micro-exceptions via `NtGetContextThread`. The process hangs indefinitely until an internal 15-minute timeout forces a thread state flush.

### Steps to Reproduce
1. Add Battle.net Installer/Launcher as a non-Steam game on Ubuntu 26.04.
2. Set launch arguments for Nvidia offloading: `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia %command%`
3. Launch the application. Observe the system hang with high micro-exception overhead in logs.

## System Information

* **GPU:** NVIDIA GeForce RTX™ 5070 Laptop GPU and AMD Radeon™ 890M Graphics
* **Driver Version:** NVIDIA-SMI 595.71.05 
* **Kernel Version:** 7.0.0-15-generic
- Link to full system information report as [Gist](https://gist.github.com/andrewsuperlegit/8d3bad2c96077f44443102b4bbdeef61


Here's the log output; i force quit it after a few minutes but i can run it for way longer if you'd like or if itd be helpful: 

[steam-13043138864154148864.log](https://github.com/user-attachments/files/28187560/steam-13043138864154148864.log)

):
- Proton version:

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

**Comment by andrewsuperlegit** [comment]
It should be noted that this https://github.com/ValveSoftware/Proton/issues/7302#issuecomment-4528696853 
was not happening in Ubuntu 25.10. Battle.net started failing only after the 26.04 upgrade. If the steps to reproduce don't cause this behavior, maybe try installing ubuntu 25.10, then installing steam and adding battle.net to steam, and confirming it works fine, and then upgradeing ubuntu to 26.04.

**Comment by ruinedraziel** [comment]
I'm noticing a high CPU usage when using 'certain windows sizes' on battle.net launcher trough proton.

Wich I can solve it by resizing it just a little. 
Screenshots

<img width="1912" height="1060" alt="Image" src="https://github.com/user-attachments/assets/be13bb0a-0ccc-4ec4-8a3e-9116e2886db5" />
<img width="1920" height="1059" alt="Image" src="https://github.com/user-attachments/assets/4b7f88d0-fcca-4a28-af05-8f69cb8de2ae" />

Small video

https://github.com/user-attachments/assets/9cc9d913-8ca4-4f3f-805a-b9c4d9c0f9d1

This is not reproducible in KDE Plasma, might be a Hyprland specific.

Hyprland 0.55.4 (Wayland)
CachyOS
Linux 7.1.1-2-cachyos
Mesa-Git 26.2.0-devel (git-05a6b192a5)
